File: packages/contracts/src/organization-member.ts
import type { ModuleCode } from "./organization-module";
import type { OrganizationMembershipRecord, OrganizationMembershipRoleCode } from "./organization-membership";
import type { ModuleAccessLevel } from "./rbac";
import type { UserRecord, UserStatus } from "./user";

export interface OrganizationMemberModuleAccessRecord {
  module_code: ModuleCode;
  module_label: string;
  access_level: ModuleAccessLevel;
  access_label: string;
  is_enabled: boolean;
}

export interface OrganizationMemberRecord {
  membership: OrganizationMembershipRecord;
  user: UserRecord;
  role_label: string;
  role_summary: string;
  access_overview: string;
  module_access: OrganizationMemberModuleAccessRecord[];
  team_ids: string[];
  team_names: string[];
}

export interface OrganizationMemberCreateRequest {
  email: string;
  first_name: string;
  last_name: string;
  phone?: string | null;
  role_code: OrganizationMembershipRoleCode;
}

export interface OrganizationMemberUpdateRequest {
  role_code?: OrganizationMembershipRoleCode | null;
  user_status?: UserStatus | null;
}

File: apps/api/app/schemas/organization_member.py
from uuid import UUID

from pydantic import BaseModel, ConfigDict

from app.db.models.organization_module import OrganizationModuleCode
from app.db.models.user import UserStatus
from app.schemas.organization_membership import OrganizationMembershipRead
from app.schemas.user import UserRead


class OrganizationMemberModuleAccessRead(BaseModel):
    module_code: OrganizationModuleCode
    module_label: str
    access_level: str
    access_label: str
    is_enabled: bool


class OrganizationMemberRead(BaseModel):
    membership: OrganizationMembershipRead
    user: UserRead
    role_label: str
    role_summary: str
    access_overview: str
    module_access: list[OrganizationMemberModuleAccessRead]
    team_ids: list[UUID]
    team_names: list[str]

    model_config = ConfigDict(from_attributes=True)


class OrganizationMemberCreateRequest(BaseModel):
    email: str
    first_name: str
    last_name: str
    phone: str | None = None
    role_code: str


class OrganizationMemberUpdateRequest(BaseModel):
    role_code: str | None = None
    user_status: UserStatus | None = None
